mdal/frmts/mdal_hec2d.cpp

```diff
@@ -398,7 +407,16 @@ std::shared_ptr<MDAL::MemoryDataset2D> MDAL::DriverHec2D::readElemOutput( const
     std::string flowAreaName = flowAreaNames[nArea];
     HdfGroup gFlowAreaRes = openHdfGroup( rootGroup, flowAreaName );
 
-    HdfDataset dsVals = openHdfDataset( gFlowAreaRes, rawDatasetName );
+    HdfDataset dsVals;
+    try
+    {
+      dsVals = openHdfDataset( gFlowAreaRes, rawDatasetName );
+    }
+    catch ( MDAL::Error )
+    {
+      return nullptr;
```

Comment:
function returns void

Reply:
std::shared_ptr<MDAL::MemoryDataset2D> is not void.
but make think it will be better to throw a exception outside this method if this is bed elevation